### PR TITLE
Updated docs for jdt.ls integration

### PR DESCRIPTION
**Pull request by JPinkney**
Signed-off-by: jpinkney <josh.pinkney@mail.utoronto.ca>

Updated documentation for jdt.ls integration in all applicable places (that I could find)

*Shouldn't be merged until 5730_java_ls_poc branch is merged in master to avoid confusion*
Compantion PR: https://github.com/eclipse/che/pull/10863

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Review by tsmaeder — DISMISSED** [review]
ltgm

**Comment by tsmaeder** [comment]
We must not merge this one until branch 5730_java_ls_poc is merged.

**Review by tsmaeder — CHANGES_REQUESTED** [review]
ltgm, but wait until we merge branch 5730_java_ls_poc

**Review by slemeur — APPROVED** [review]

**Review by tsmaeder — APPROVED** [review]

**Pull request merged**
